caikit/core/model_management/local_model_trainer.py

```diff
@@ -173,14 +173,7 @@ def wait(self):
 
             # If running a subprocess, handle abnormal exit codes
             if self._use_subprocess:
-                self._completion_time = self._worker.get_or_throw()
-                if self._worker.exitcode and self._worker.exitcode != os.EX_OK:
-                    if self._worker.exitcode == OOM_EXIT_CODE:
-                        raise MemoryError("Training process died with OOM error!")
-                    if not self._worker.canceled:
-                        raise RuntimeError(
-                            f"Training process died with exit code {self._worker.exitcode}"
-                        )
+                self._completion_time = datetime.now()
```

Comment:
What happened to getting this from the child? Also, even if we go with this, we need to make sure it's only set the first time. This would also allow you to remove the `if self._use_subprocess` check.

```py
self._completion_time = self._completion_time or datetime.now()
```

Reply:
Also, if we do go with this, you can remove the janky bits about returning the completion time in `_train_and_save`

Reply:
> What happened to getting this from the child?

I thought doing it this way would be mostly equivalent.

Reply:
Another problem with get_or_throw is, that function will throw error, in which cases, we won't get the completion time set.. which could be a problem?